src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -247,6 +247,26 @@ QgsPostgresProvider::QgsPostgresProvider( QString const &uri, const ProviderOpti
     }
   }
 
+  // check if geometry is 3D or not
+  if ( !mGeometryColumn.isEmpty() && !mTableName.isEmpty() )
+  {
+    const QString selectQuery = QStringLiteral( R"SQL(
```

Comment:
Again, why not just rely on the wkb type here?

Reply:
fixed

Reply:
We will reduce readability because the schemas/tables are not the same. This request is made once, the impact will be low.

Reply:
I respectfully disagree -- see eg all the work done by https://github.com/qgis/QGIS/pulls?q=is%3Apr+is%3Aclosed+author%3Amorriganr focused on removing every possible extraneous query. That's driven by real world requirements.

Reply:
Ok but I don't know how to properly do this merged query. Maybe @MorriganR could help us find a solution (please :grin:)?

Reply:
Can we drop the postgres changes for now then, and implement in a specific follow up PR? I don't want to see speed regressions introduced here.

Reply:
Yes it is the best solution for now